src/components/v5/common/ActionSidebar/partials/Motions/steps/StakingStep/hooks.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+
+import { BigNumber } from 'ethers';
+import { ColonyMotionFragment, useGetUserReputationQuery } from '~gql';
+import { useAppContext, useEnoughTokensForStaking } from '~hooks';
+import { formatText } from '~utils/intl';
+import { UserInfoListItem } from '~v5/shared/UserInfoSectionList/partials/UserInfoList/types';
+import { useMotionContext } from '../../partials/MotionProvider/hooks';
+import Numeral from '~shared/Numeral';
+import useUsersByAddresses from '~hooks/useUsersByAddresses';
+
+export const useStakingStep = () => {
+  const { motionAction } = useMotionContext();
+  const { user, userLoading, walletConnecting } = useAppContext();
+
+  const { colony, motionData } = motionAction;
+  const { nativeToken, colonyAddress } = colony;
+  const { tokenAddress } = nativeToken;
+
+  const { userMinStake, rootHash, nativeMotionDomainId } = motionData;
+
+  const {
+    hasEnoughTokens: enoughTokensToStakeMinimum,
+    loadingUserTokenBalance,
+    userActivatedTokens,
+  } = useEnoughTokensForStaking(
+    tokenAddress,
+    user?.walletAddress ?? '',
+    colonyAddress,
+    userMinStake,
+  );
+
+  const { data, loading: loadingReputation } = useGetUserReputationQuery({
+    variables: {
+      input: {
+        colonyAddress,
+        walletAddress: user?.walletAddress ?? '',
+        domainId: Number(nativeMotionDomainId),
+        rootHash,
+      },
+    },
+  });
+
+  const userReputation = data?.getUserReputation;
+  /* User cannot stake more than their reputation in tokens. */
+  const userMaxStake = BigNumber.from(userReputation ?? '0');
+
+  const enoughReputationToStakeMinimum =
+    userMaxStake.gt(0) && userMaxStake.gte(userMinStake);
+
+  return {
+    isLoading:
+      userLoading ||
+      walletConnecting ||
+      loadingUserTokenBalance ||
+      loadingReputation,
+    enoughTokensToStakeMinimum,
+    enoughReputationToStakeMinimum,
+    userActivatedTokens,
+  };
+};
+
+export const useStakingInformation = (
+  usersStakes: ColonyMotionFragment['usersStakes'],
+  tokenDecimals: number,
+  tokenSymbol: string,
+): {
+  votesFor: UserInfoListItem[];
+  votesAgainst: UserInfoListItem[];
+  isLoading?: boolean;
+} => {
+  const { users, loading } = useUsersByAddresses(
+    usersStakes.map((user) => user.address),
+  );
+
+  const votesFor: UserInfoListItem[] =
+    usersStakes?.reduce((result, item) => {
+      if (!item || item.stakes?.raw?.yay === '0') {
+        return result;
+      }
+
+      const userName = users?.length
+        ? users?.find((user) => {
+            const { walletAddress } = user || {};
+            return walletAddress === item.address;
+          })?.profile?.displayName
```

Comment:
can you write this condition in a single line?

Reply:
What do you mean?